FF153 Popover API docs for new hint/auto popover interaction model (#44603)
mdn/content: files/en-us/web/api/popover_api/using/index.md

});
```

> Note that calling {{domxref("HTMLElement.showPopover()", "showPopover()")}}, {{domxref("HTMLElement.hidePopover()", "hidePopover()")}}, or {{domxref("HTMLElement.togglePopover()", "togglePopover()")}} from within a `beforetoggle` or `toggle` event handler while another popover is already being shown or hidden is not permitted, and will throw an `InvalidStateError` `DOMException`.

Reviewer: ```suggestion
Note that calling {{domxref("HTMLElement.showPopover()", "showPopover()")}}, {{domxref("HTMLElement.hidePopover()", "hidePopover()")}}, or {{domxref("HTMLElement.togglePopover()", "togglePopover()")}} from within a `beforetoggle` event listener while another popover is already being shown or hidden is not permitted, and will throw an `InvalidStateError` `DOMException`.
```

   <div popover anchor="foo">Child</div>
   ```

> See our [Nested popover menu example](https://mdn.github.io/dom-examples/popover-api/nested-popovers/) ([source](https://github.com/mdn/dom-examples/tree/main/popover-api/nested-popovers)) for an example. You'll notice that quite a few event handlers have been used to display and hide the subpopover appropriately during mouse and keyboard access, and also to hide both menus when an option is selected from either. Depending on how you handle loading of new content, either in an SPA or multi-page website, some or all of these may not be necessary, but they have been included in this demo for illustrative purposes.

Author: This stuff is just moved - it was nested at the wrong place. Nothing to do with this PR really.

</div>
```

## Using "hint" popover state

There is a third type of popover you can create — **hint popovers**, designated by setting `popover="hint"` on your popover element.
They can be light dismissed and will respond to close requests.

`hint` popovers do not close `auto` popovers when they are displayed, but will close other `hint` popovers that are not its ancestors in the [hint stack](#popover_openclose_interaction_rules).

> The reverse is also true: closing an `auto` popover — whether by showing another `auto`, pressing <kbd>Esc</kbd>, or light-dismiss — does not affect `hint` popovers unless they are descendants of the closed auto popover.

Reviewer: ```suggestion
The reverse is also true: closing an `auto` popover by pressing <kbd>Esc</kbd> or light-dismiss does not affect `hint` popovers unless they are descendants of the closed auto popover.
```

Showing another `auto` would close the hint.

Author: Thanks, I've modified your version very slightly, because keeping "whether" still carries the possible implication that any mechanism (including showing another auto` will not close the hint.

}
```

## Popover open/close interaction rules

The browser maintains two independent stacks of open popovers: an **auto stack** for `auto` popovers, and a **hint stack** for `hint` popovers.
When a popover is shown it is pushed onto the appropriate stack; when it is hidden, the browser walks back down that stack, closing any descendant popovers on that stack first.
Because the two stacks are separate, operations on one do not automatically affect the other.

A few specific rules for how popovers interact that derive from this specification are:

- Showing a `hint` popover does not close `auto` popovers.
- Showing a `hint` popover closes other `hint` popovers, except those that are its ancestors in the hint stack.

> - Clicking outside a popover light-dismisses all open `auto` and `hint` popovers. <!-- TODO: What happens if you click outside a hint, but inside the parent? -->

Reviewer: > What happens if you click outside a hint, but inside the parent?

The hint will close, but the parent will not.

Although the hint & auto stacks are kinda different, the hint stack can be rooted to an auto popover, like this:

```mermaid
graph TD
    auto_1[auto 1] --> auto_2[auto 2]
    auto_2 --> auto_3[auto 3]
    auto_2 --> hint_1[hint 1]
    hint_1 --> hint_2[hint 2]
```

This happens if the hint was opened as a child of an auto. If it isn't, then the two stacks are separate. 

In the case above, when you click one of those popovers, it and its parents will remain, but the others will hide. So if you click "auto 3", the hints will hide. If you click "hint 1", then "auto 3" and "hint 2" will hide.

Author: Thanks. I'm going to modify this to:

```suggestion
- Clicking outside a popover light-dismisses all open `auto` and `hint` popovers that are not its ancestor.
```

- Hiding an `auto` popover does not close `hint` popovers that are not its descendants.
- Showing an `auto` popover as a child of a `hint` popover downgrades the `auto` popover to `hint`.
- Showing a popover while another is in the process of being shown or hidden is not permitted.

> Note that `manual` popovers do not participate in either stack — they are shown and hidden independently and have no effect on the auto or hint stack.

Reviewer 2: ```suggestion
Note that `manual` popovers do not participate in either stack — they are shown and hidden independently and do not affect auto or hint popovers.
```

Maybe, just to get rid of "stack" repetition?

> 

Author: @jakearchibald This is the new section - can you please sanity check - this is my understanding + claudes. 

Note that in (my) ideal world this would be in the API overview along with a description of the different popover types (rather than this info all being in the "using guide". However I don't want to force a major restructure at this point.
